src/04_SearchingAlgrothims/SearchingAlgrothims.c

```diff
@@ -0,0 +1,39 @@
+#include "SearchingAlgrothims.h"
+Element_Type RecBinarySearch(Element_Type arr[],Element_Type number,Element_Type first,Element_Type last)
+{
+	while(first<=last)
+   {
+	Element_Type mid;
```

Comment:
it's preferred to add suffix `index` to note the difference between the value and index i.e  `Element_Type mid_index`